step-plans/step-plans-base-artefacts/src/main/java/step/artefacts/handlers/ForBlockHandler.java

```diff
@@ -152,7 +153,7 @@ public Consumer<DataPoolRow> createWorkItemConsumer(WorkerController<DataPoolRow
 						}
 					};
 				}
-			}, numberOfThreads);
+			}, numberOfThreads, OptionalInt.empty());

```

Comment:
Would recommend to add a new method consumeWork with the additional parameter in order to keep the signature and avoid this change here